vm/src/actor_state.rs

```diff
@@ -43,3 +43,79 @@ impl ActorState {
         self.balance += amt;
     }
 }
+
+#[cfg(feature = "json")]
+pub mod json {
+    use super::*;
+    use crate::TokenAmount;
+    use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+    use std::str::FromStr;
+    /// Wrapper for serializing and deserializing a SignedMessage from JSON.
+    #[derive(Deserialize, Serialize)]
+    #[serde(transparent)]
+    pub struct ActorStateJson(#[serde(with = "self")] pub ActorState);
+
+    /// Wrapper for serializing a SignedMessage reference to JSON.
+    #[derive(Serialize)]
+    #[serde(transparent)]
+    pub struct ActorStateJsonRef<'a>(#[serde(with = "self")] pub &'a ActorState);
+
+    impl From<ActorStateJson> for ActorState {
+        fn from(wrapper: ActorStateJson) -> Self {
+            wrapper.0
+        }
+    }
+
+    pub fn serialize<S>(m: &ActorState, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        #[derive(Serialize)]
+        #[serde(rename_all = "PascalCase")]
+        struct ActorStateSer<'a> {
+            #[serde(with = "cid::json")]
+            pub code: &'a Cid,
+            #[serde(rename = "Head", with = "cid::json")]
+            pub state: &'a Cid,
+            #[serde(rename = "Nonce")]
+            pub sequence: u64,
+            pub balance: &'a String,
+        }
+        ActorStateSer {
+            code: &m.code,
+            state: &m.state,
+            sequence: m.sequence,
+            balance: &m.balance.to_str_radix(10),
+        }
+        .serialize(serializer)
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<ActorState, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        #[derive(Serialize, Deserialize)]
```

Comment:
```suggestion
        #[derive(Deserialize)]
```
Again doesn't really matter but no need since it's private and only used as a helper struct